test/fig-completion.cjs

```diff
@@ -0,0 +1,1375 @@
+          .parse()
+      );
+      o.exitCode.should.equal(0);
+      o.errors.length.should.equal(0);
+      o.logs[0]
+        .split('\n')
+        .should.deep.equal([
+          '// Autogenerated by the Yargs integration',
+          '',
+          'const completionSpec: Fig.Spec = {',
+          '  "name": [',
+          '    "usage"',
+          '  ],',
+          '  "description": "Default",',
+          '  "options": [',
+          '    {',
+          '      "name": [',
+          '        "--bopt"',
+          '      ],',
+          '      "args": {',
+          '        "name": "boolean",',
+          '        "suggestions": [',
+          '          "true",',
+          '          "false"',
+          '        ],',
+          '        "isOptional": true',
+          '      }',
+          '    }',
+          '  ],',
+          '  "subcommands": []',
+          '};',
+          '',
+          'export default completionSpec;',
+          '',
+        ]);
+    });
+  });
+
+  it('.showFigCompletion() should work without parsing', () => {
+    const o = checkOutput(() =>
+      yargs()
+        .command(
+          '$0',
+          'An awesome description',
+          y => {
+            return y.option('delta').alias('delta', 'd').boolean('delta');
+          },
+          () => {}
+        )
+        .showFigCompletion()
+    );
+    o.exitCode.should.equal(0);
+    o.errors.length.should.equal(0);
+    o.logs[0]
+      .split('\n')
+      .should.deep.equal([
+        '// Autogenerated by the Yargs integration',
+        '',
+        'const completionSpec: Fig.Spec = {',
+        '  "name": [',
+        '    "usage"',
+        '  ],',
+        '  "description": "An awesome description",',
+        '  "subcommands": [],',
+        '  "options": [',
+        '    {',
+        '      "name": [',
+        '        "--help"',
+        '      ],',
+        '      "description": "Show help"',
+        '    },',
+        '    {',
+        '      "name": [',
+        '        "--version"',
+        '      ],',
+        '      "description": "Show version number"',
+        '    },',
+        '    {',
+        '      "name": [',
+        '        "--delta",',
+        '        "-d"',
+        '      ],',
+        '      "exclusiveOn": [',
+        '        "--no-delta"',
+        '      ],',
+        '      "args": {',
+        '        "name": "boolean",',
+        '        "suggestions": [',
+        '          "true",',
+        '          "false"',
+        '        ],',
+        '        "isOptional": true',
+        '      }',
+        '    },',
+        '    {',
+        '      "name": [',
+        '        "--no-delta",',
+        '        "--no-d"',
+        '      ],',
+        '      "exclusiveOn": [',
+        '        "--delta"',
+        '      ]',
+        '    }',
+        '  ]',
+        '};',
+        '',
+        'export default completionSpec;',
+        '',
+      ]);
+  });
+
+  // TODO(fedeci): not working properly
```

Comment:
This change is not required by fig right now so we can keep it for a second PR. 
It involves getting the correct yargs "subinstance" and I found that pretty tricky. If anyone can point me in the right direction I can eventually make the change directly in this PR.